File: chapter2/fizz_buzz.js
for(let n = 1; n <= 100; n++) {
  if (n % 15 === 0) {
    console.log('FizzBuzz')
  } else if (n % 3 === 0) {
    console.log('Fizz') 
  } else if (n % 5 === 0) {
    console.log('Buzz')
  } else {
    console.log(`${n}`)
  }
}

// 模範解答
for (let n = 1; n <= 100; n++) {
  let output = "";
  if (n % 3 == 0) output += "Fizz";
  if (n % 5 == 0) output += "Buzz";
  console.log(output || n);
}

// 始まりと終わりを指定できるようにした
const fizzBuzz = (start, end) => {
  for(let n = start; n <= end; n++) {
    if (n % 15 === 0) {
      console.log('FizzBuzz')
    } else if (n % 3 === 0) {
      console.log('Fizz') 
    } else if (n % 5 === 0) {
      console.log('Buzz')
    } else {
      console.log(`${n}`)
    }
  }
}

File: chapter2/fizz_buzz.rb
(1..100).each do |n|
  puts output =
    if n % 15 == 0
      'FizzBuzz'
    elsif n % 3 == 0
      'Fizz'
    elsif n % 5 == 0
      'Buzz'
    else
      n.to_s
    end
end

# 始まりと終わりを指定できるようにした
def fizz_buzz(start, last)
  (start..last).each do |n|
    puts output =
      if n % 15 == 0
        'FizzBuzz'
      elsif n % 3 == 0
        'Fizz'
      elsif n % 5 == 0
        'Buzz'
      else
        n.to_s
      end
  end
end


